### ISSUE TITLE
ReorderableWrap - disable reordering for specific items

### ISSUE THREAD

**Issue by e-e** [hanshengchiu/reorderables#89]
Is it possible to disable drag for specific items in the `ReorderableWrap` class? My specific use case is to create a photo upload screen that allows a user to upload a set number of images, building up a grid as they add photos. I would like the last item in the grid to be a "+" button used for adding the next image, but i do not want them to be able to reorder this item.

**Comment by Victor-Dev77** [comment]
I've the same problem for the same use...

**Comment by peakerWu** [comment]
I've the same problem for the same use.

**Comment by e-e** [comment]
In the meantime, in case that can help someone else, this is roughly my workaround:

```
class MyWidget extends StatelessWidget {
   List<MyGridItemWidget> gridItems = [];

  // ...

  @override
  Widget build(BuildContext context) {
    this.gridItems = this.gridItems.where((element) => !element.isButton).toList();
    this.gridItems.add(MyButtonWidget());

    return ReorderableWrap(children: this.gridItems);
  }
}

```

The button is still draggable, but will always be render in the last position.

**Comment by Kal-Elx** [comment]
@e-e I'm doing the same thing.

I'm solving this by wrapping the non movable widgets in:
```
GestureDetector(
      onLongPress: () {}, // Override onLongPress to make item unmovable.
      child: ...
)
```

**Comment by diickens** [comment]
@Kal-Elx good workaround.. but the problem is the Tile is still reordable (move another tile on its place) would be nice to have the ability to deactivate that.

**Comment by Kal-Elx** [comment]
@diickens Yes, my solution was to enforce that in `onReorder`. It's quite easy if you just want to ensure that tiles can't be moved beyond the unmovable tile like in a photo upload screen. 

Here's how I implemented it. You can also take a look at the rest of the widget for inspiration on a photo upload screen.

https://github.com/dibbs-team/dibbs/blob/main/lib/widgets/upload/image_sequence_picker/image_sequence_picker.dart#L90-L99

**Comment by mattia-venturini** [comment]
If the item that should not be draggable is the first or the last you can use either the parameter `header` or `footer`. They seem to work fine.

**Comment by 3111279132** [comment]
> @e-e I'm doing the same thing.
> 
> I'm solving this by wrapping the non movable widgets in:
> 
> ```
> GestureDetector(
>       onLongPress: () {}, // Override onLongPress to make item unmovable.
>       child: ...
> )
> ```
This is available; thanks

**Comment by KuuBee** [comment]
> @e-e I'm doing the same thing.
> 
> I'm solving this by wrapping the non movable widgets in:
> 
> ```
> GestureDetector(
>       onLongPress: () {}, // Override onLongPress to make item unmovable.
>       child: ...
> )
> ```

This can also take effect
```dart
ReorderableWrap(
      children: [
        ReorderableWidget(
          key: ValueKey('Disable drag'),
          reorderable: false,
          child: Text('Disable drag'),
        ),
        ReorderableWidget(
          key: ValueKey('Enable drag'),
          reorderable: true,
          child: Text('Enable drag'),
        ),
      ],
      needsLongPressDraggable: false,
      onReorder: _onReorder,
    );
```

**Comment by yoniker** [comment]
> > @e-e I'm doing the same thing.
> > I'm solving this by wrapping the non movable widgets in:
> > ```
> > GestureDetector(
> >       onLongPress: () {}, // Override onLongPress to make item unmovable.
> >       child: ...
> > )
> > ```
> 
> This can also take effect
> 
> ```dart
> ReorderableWrap(
>       children: [
>         ReorderableWidget(
>           key: ValueKey('Disable drag'),
>           reorderable: false,
>           child: Text('Disable drag'),
>         ),
>         ReorderableWidget(
>           key: ValueKey('Enable drag'),
>           reorderable: true,
>           child: Text('Enable drag'),
>         ),
>       ],
>       needsLongPressDraggable: false,
>       onReorder: _onReorder,
>     );
> ```

This is the most native solution I can see, but still this allows to change the position of non-draggables by dragging over, and the footer widget doesn't obey the same placement rules as the other widgets (for example in a Wrap it's "after" the wrap eg underneath instead of whereever the next element would have been in a wrap).

 @diegotori I'm facing the exact same scenario and I can see that this issue is open for a while, is it a huge hassle to implement a "locked" item which obeys the same rules as the original Widget(eg Wrap,Column,Row etc)?

**Comment by cgowthamanmca** [comment]
> > @e-e I'm doing the same thing.
> > I'm solving this by wrapping the non movable widgets in:
> > ```
> > GestureDetector(
> >       onLongPress: () {}, // Override onLongPress to make item unmovable.
> >       child: ...
> > )
> > ```
> 
> This can also take effect
> 
> ```dart
> ReorderableWrap(
>       children: [
>         ReorderableWidget(
>           key: ValueKey('Disable drag'),
>           reorderable: false,
>           child: Text('Disable drag'),
>         ),
>         ReorderableWidget(
>           key: ValueKey('Enable drag'),
>           reorderable: true,
>           child: Text('Enable drag'),
>         ),
>       ],
>       needsLongPressDraggable: false,
>       onReorder: _onReorder,
>     );
> ```

Thanks lot

**Comment by MarnoDev** [comment]
> > @e-e I'm doing the same thing.
> > I'm solving this by wrapping the non movable widgets in:
> > ```
> > GestureDetector(
> >       onLongPress: () {}, // Override onLongPress to make item unmovable.
> >       child: ...
> > )
> > ```
> 
> This can also take effect
> 
> ```dart
> ReorderableWrap(
>       children: [
>         ReorderableWidget(
>           key: ValueKey('Disable drag'),
>           reorderable: false,
>           child: Text('Disable drag'),
>         ),
>         ReorderableWidget(
>           key: ValueKey('Enable drag'),
>           reorderable: true,
>           child: Text('Enable drag'),
>         ),
>       ],
>       needsLongPressDraggable: false,
>       onReorder: _onReorder,
>     );
> ```

Not work
